deepchecks/nlp/checks/data_integrity/under_annotated_segments.py

```diff
@@ -0,0 +1,197 @@
+from deepchecks.nlp.utils.weak_segments import get_relevant_data_table
+from deepchecks.utils.abstracts.weak_segment_abstract import WeakSegmentAbstract
+from deepchecks.utils.strings import format_percent
+from deepchecks.utils.typing import Hashable
+
+__all__ = ['UnderAnnotatedMetaDataSegments', 'UnderAnnotatedPropertySegments']
+
+
+class UnderAnnotatedSegments(SingleDatasetCheck, WeakSegmentAbstract):
+    """Check for under annotated data segments."""
+
+    def __init__(self, segment_by: str, columns: Union[Hashable, List[Hashable], None],
+                 ignore_columns: Union[Hashable, List[Hashable], None], n_top_features: int,
+                 segment_minimum_size_ratio: float, n_samples: int,
+                 categorical_aggregation_threshold: float, n_to_show: int, **kwargs):
+        super().__init__(**kwargs)
+        self.segment_by = segment_by
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.n_top_features = n_top_features
+        self.segment_minimum_size_ratio = segment_minimum_size_ratio
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.categorical_aggregation_threshold = categorical_aggregation_threshold
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        context.raise_if_token_classification_task(self)
+        context.raise_if_multi_label_task(self)
+
+        text_data = context.get_data_by_kind(dataset_kind)
+        text_data = text_data.sample(self.n_samples, random_state=context.random_state)
+
+        features, cat_features = get_relevant_data_table(text_data, data_type=self.segment_by,
+                                                         columns=self.columns, ignore_columns=self.ignore_columns,
+                                                         n_top_features=self.n_top_features)
+
+        encoded_dataset = self._target_encode_categorical_features_fill_na(features, text_data.label,
+                                                                           cat_features)
+
+        loss_per_sample = pd.Series([1 - pd.isna(x) for x in text_data.label], index=encoded_dataset.data.index)
+        weak_segments = self._weak_segments_search(data=encoded_dataset.features_columns,
+                                                   loss_per_sample=loss_per_sample,
+                                                   scorer_name='Annotation Ratio')
+
+        if len(weak_segments) == 0:
+            raise DeepchecksProcessError('Check was unable to find under annotated segments. This is expected if '
+                                         'your data is well annotated. If this is not the case, try increasing '
+                                         f'n_samples or supply more {self.segment_by}.')
+
+        avg_score = round(loss_per_sample.mean(), 3)
+
+        weak_segments = self._generate_check_value_display(weak_segments, cat_features)
```

Comment:
P0 here (prior to the scatter plot) can be the existing plot with a different color scheme.

Reply:
mm good idea, lets talk